docs/metadata.md

```diff
@@ -273,6 +274,11 @@ Note that this does not apply to replacement mechanisms supported natively by so
 package managers, such as the go.mod [replace][gomod-replace] directive. Those
 replacements will not be reflected in the request JSON in any way.
 
+#### dependency.kind
+
+Returned only for `RubyGems` dependencies, this makes it easy to distinguish between different 
+kinds of them. Possible values are `GEM`, `GIT` or `PATH`. 
```

Comment:
I feel like this sets up for more problems in the future. It's not very well justifiable why this key is only used for ruby when it's also applicable to pip, npm, yarn and maybe even gomod. But if we want to add `kind` to the other package managers in the future and the known values for it are `GEM`, `GIT`, `PATH`, it will make things awkward.

The rest of the Cachito team, what do you think?

Reply:
I could still do it the same way as `gomod` does it with PATH dependencies starting with `./`, or I could come up with some values that are universal, such as `git`, `path`, `url`, `package-repository`.

Reply:
I agree here with @chmeliik. In case we can't use the same set of values [GIT, GEM, PATH] for all package managers with the `kind` item, we should rather use the gomod approach and distinguish `kind/type` from parsing version string. @tichavskym do we have a set of all possible versions to be resolved here as GIT, GEM, PATH "kind" which we could add to unit tests?

Reply:
> The RubyGems team urges gem developers to follow the [Semantic Versioning](http://semver.org/) standard for their gem’s versions. The RubyGems library itself does not enforce a strict versioning policy, but using an “irrational” policy will only be a disservice to those in the community who use your gems.

In other words, GEM dependency can have whatever it wants as a version. So, we could say GIT dependencies' version will be in format `git+<url>@hash` and PATH dependencies' version will start with `.` [as gomod is doing](https://github.com/containerbuildsystem/cachito/blob/28a6afc130df5296945898b7d8578e5bf2157ac0/cachito/web/content_manifest.py#L322) or `path+` if we wanted to come up with something new, but there's no guarantee that it won't collide with any of the GEM versions, so we would have to hope maintainers will name it so that it makes sense.

Reply:
Meaning, for every GEM dependency with version starting with `.` (which is possible to have), we would produce the wrong purl. If we used `kind` internally and removed it from the responses, the purl would be produced correctly, but it would still be confusing in the response JSON, because users would look at version starting with `.` and be thinking that its a PATH dependency. The same would probably be true for all GEM dependencies with versions starting with `git+`. I'm not sure to which degree we want to take care of these edge cases (or what other package managers are doing about this).

Reply:
I wouldn't worry too much about these edge cases. Maybe check that PATH deps start with `./` -- if someone uses a version like that for their GEM, I'd assume they're deliberately messing with us.

I think pip isn't all that strict about enforcing the version rules either.

And you don't even want to know what we do with NPM, the purls are not even close to valid and haven't been for 2 years :sweat_smile:

Reply:
Ok :laughing: In that case, this pull request is ready for merge/final review.